pyspeckit/spectrum/units.py

```diff
@@ -453,83 +400,64 @@ def __array_wrap__(self,out_arr,context=None):
             pass
         return ret
 
-    def __repr__(self):
-        if self.shape is ():
-            rep = ("SpectroscopicAxis([%r], units=%r, refX=%r, refX_units=%r, frame=%r, redshift=%r, xtype=%r, velocity convention=%r)" %
-                (self.__array__(), self.units, self.refX, self.refX_units, self.frame, self.redshift, self.xtype, self.velocity_convention))
-        else:
-            rep = ("SpectroscopicAxis([%r,...,%r], units=%r, refX=%r, refX_units=%r, frame=%r, redshift=%r, xtype=%r, velocity convention=%r)" %
-                (self[0], self[-1], self.units, self.refX, self.refX_units, self.frame, self.redshift, self.xtype, self.velocity_convention))
-        return rep
-
-    def __str__(self):
-        selfstr =  "SpectroscopicAxis with units %s and range %g:%g." % (
-                self.units,self.umin(),self.umax())
-        if self.refX is not None:
-            selfstr += "Reference is %g %s" % (self.refX, self.refX_units)
-        return selfstr
-
-    @property
-    def unit(self):
-        return self._unit
-
-    @unit.setter
-    def unit(self, value):
-        self._unit = value
-
-    @property
-    def units(self):
-        return self.unit
-
-    @units.setter
-    def units(self, value):
-        self.unit = value
-
-    def _plural_sanity_check(self):
-        if hasattr(self,'unit') and hasattr(self,'units'):
-            assert self.unit == self.units
-
     def _check_consistent_type(self):
         """
-        Make sure self.xtype is unit_type_dict[units]
+        Make sure self.xtype is unit_type_dict[unit]
         if this is NOT true, can cause significant errors
         """
-        self._plural_sanity_check()
         if self.xtype is None:
             OK = False
-        elif self.unit not in unit_type_dict:
-            raise KeyError("Unit {0} is not recognized.".format(self.unit))
         else:
-            OK = self.xtype.lower() == unit_type_dict[self.unit]
+            OK = self.xtype.lower() == self.unit
 
         if not OK:
-            raise InconsistentTypeError("Units: %s Type[units]: %s in %r" % (self.unit,unit_type_dict[self.unit],self) )
+            raise InconsistentTypeError("Unit: %s Type[unit]: %s in %r" % (self.unit,unit_type_dict[self.unit],self))
 
     def _fix_type(self):
         try:
             self._check_consistent_type()
         except InconsistentTypeError:
-            self.xtype = unit_type_dict[self.units]
+            self.xtype = unit_type_dict[self.unit]
 
-    def umax(self, units=None):
+    @classmethod
+    def validate_unit(self, unit, bad_unit_response='raise'):
+        try:
+            if unit is None or unit == 'unknown':
+                unit = u.dimensionless_unscaled
+            if unit == 'angstroms': unit = 'angstrom'
+            unit = u.Unit(unit)
+        except ValueError:
+            if bad_unit_response == "pixel":
+                unit = "unknown"
```

Comment:
should this just be `u.dimensionless_unscaled` since you're setting it that way on line 426?